### PR TITLE
CLOUDP-67935: create project api key

### PR DESCRIPTION
**Pull request by robcarlan-mlab**
## Proposed changes

_Jira ticket:_ CLOUDP-67935

Closes #[issue number]

## Checklist

- [] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [] I have added tests that prove my fix is effective or that my feature works
- [] I have added any necessary documentation (if appropriate)
- [] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli iam projects apikeys create --description a --role GROUP_READ_ONLY
New API Key '5f231e4843bb36438fef1d34' created for project.

./bin/mongocli iam projects apikeys create --description b --role GROUP_READ_ONLY -o json
{
  "id": "5f231e54f2828157f640b001",
  "desc": “b”,
  "roles": [
    {
      "orgId": "5cac6a2179358edabd12b572",
      "roleName": "ORG_READ_ONLY"
    },
    {
      "groupId": "5cac6a2ba6f239298a18a025",
      "roleName": "GROUP_READ_ONLY"
    }
  ],
  "privateKey": "94446e1e-2c92-42f7-a012-c06e36305fe5",
  "publicKey": "kkromxsv"
}
```

### CODE REVIEW

File: internal/usage/usage.go [hunk 1]
```diff
@@ -71,6 +71,8 @@ const (
 	NotificationToken               = "Slack API token or Bot token or Flowdock personal API token." //nolint:gosec // This is just a message not a password
 	NotificationsChannelName        = "Slack channel name. Required for the SLACK notifications type."
 	APIKey                          = "Datadog API Key, Opsgenie API Key, VictorOps API key."
+	APIKeyDescription               = "Description of the API key."
```
Comment (andreaangiolillo) [inline internal/usage/usage.go:74]: I think that for description you can reuse `usage.Comment`

File: internal/usage/usage.go [hunk 2]
```diff
@@ -71,6 +71,8 @@ const (
...
+	APIKeyRoles                     = "List of roles for the API key."
```
Comment (andreaangiolillo) [inline internal/usage/usage.go:75]: We can reuse `usage.Roles` here

File: internal/cli/projects/apikeys/create.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		},
+	}
+
+	cmd.Flags().StringArrayVar(&opts.roles, flag.Role, nil, usage.APIKeyRoles)
+	cmd.Flags().StringVar(&opts.description, flag.Description, "", usage.APIKeyDescription)
+
+	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
+
+	_ = cmd.MarkFlagRequired(flag.Description)
+	_ = cmd.MarkFlagRequired(flag.Role)
```
[the remark this answers is not in the document]
Comment (robcarlan-mlab) [reply internal/cli/projects/apikeys/create.go:83]: I need to follow up on docs with this one, in practice they ended up both being required:
```
> ./bin/mongocli iam projects apikeys create --role GROUP_READ_ONLY
Error: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/5cac6a2ba6f239298a18a025/apiKeys: 400 (request "Bad Request") All API keys require a non-empty description.

> ./bin/mongocli iam projects apikeys create --description description
Error: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/5cac6a2ba6f239298a18a025/apiKeys: 500 (request "Internal Server Error") Unexpected error.

> ./bin/mongocli iam projects apikeys create --role GROUP_READ_ONLY --description description
{
  "id": "5f242f358caf14522108fcb4",
  "desc": "description",
  "roles": [
    {
      "orgId": "5cac6a2179358edabd12b572",
      "roleName": "ORG_READ_ONLY"
    },
    {
      "groupId": "5cac6a2ba6f239298a18a025",
      "roleName": "GROUP_READ_ONLY"
    }
  ],
 ...
}
```

File: internal/usage/usage.go (hunk 1 above)
Comment (robcarlan-mlab) [reply internal/usage/usage.go:74]: I'll leave this one based on the above conversation -- `description` seems to actually be mandatory

File: internal/usage/usage.go (hunk 2 above)
Comment (robcarlan-mlab) [reply internal/usage/usage.go:75]: I initially thought this too, but the `usage.Roles` refers to databases/collections

File: internal/flag/flags.go [hunk 4]
```diff
@@ -52,6 +52,7 @@ const (
 	FirstName                       = "firstName"                       // FirstName flag
 	LastName                        = "lastName"                        // LastName flag
 	Role                            = "role"                            // Role flag
+	Description                     = "description"                     // Description flag
```
[the remark this answers is not in the document]
Comment (robcarlan-mlab) [reply internal/flag/flags.go:55]: 👍  probably better to match the api, will change

File: internal/usage/usage.go (hunk 2 above)
Comment (andreaangiolillo) [reply internal/usage/usage.go:75]: we are using `usage.Roles` for creating and updating organization api keys. I think that we should update them as well if we are going to use `APIKeyRoles` here

Comment (robcarlan-mlab) [reply internal/usage/usage.go:75]: Ok i'll merge master and update those now

**Comment by robcarlan-mlab** [comment]
just updated the usage

**Pull request merged**